### PR TITLE
FIETS-520: Check geofence already in region

### PR DESCRIPTION
**Pull request by williamrijksen**
### What has been done
- Check use is already in geofence-region on start monitoring 

### How to test
- Mock location in a geofence and see the enterregion is immediately triggering. 

### Todo
- [ ] Trust on https://stackoverflow.com/questions/30282259/ios-geofence-how-to-handle-when-inside-region-when-monitoring-starts for iOS.
- [x] Optimize code

### Notes
-

### CODE REVIEW

**Comment by williamrijksen** [comment]
@jvandijk We have to consider the open todo is still required.

File: lib-ng/service/ScanService.js [hunk 1]
```diff
@@ -54,22 +55,26 @@ export default class ScanService {
...
+      regions,
+      callback,
+    };
 
-    // geofence-regions are already filtered by the hook, nearest geofences filter not needed
     if (_.isFunction(this.options.hooks.getRegionsToMonitor)) {
-      callback(regions);
-      return;
+      // Order every region because it's already filtered within the hook
+      nearestRegionsParameters.distance = 75000;
```
Comment (jvandijk) [inline lib-ng/service/ScanService.js:71]: @williamrijksen Too many zeroes?

Comment (williamrijksen) [reply lib-ng/service/ScanService.js:71]: No because the hook already filtered the regions.

File: lib-ng/service/ScanService.js [hunk 2]
```diff
@@ -54,22 +55,26 @@ export default class ScanService {
 
   startGeofence(regions) {
     const scanner = this.getGeofenceScanner();
-    const callback = nearestRegions => nearestRegions.forEach(
-      region => scanner.startMonitoring(region)
-    );
+    const callback = nearestRegions => {
+      nearestRegions.forEach((region) => scanner.startMonitoring(region));
+      // Check the use already entered the region
```
Comment (williamrijksen) [inline lib-ng/service/ScanService.js:60]: user

**Comment by jvandijk** [comment]
@williamrijksen merging, but I think this is not enough. Your assumption that on start you have the correct location is in reality not working out, so I think we have to add the open todo for iOS.

In my tests my initial location was in different devices off more then 125 meters. And then it won't trigger this logic at all.

**Pull request merged**